### PR TITLE
[MM-22572] Modify get user by email route to accept the / character in the email

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-22572

### CODE REVIEW

File: app/server.go [hunk 1]
```diff
@@ -138,6 +138,7 @@ type Server struct {
 
 func NewServer(options ...Option) (*Server, error) {
 	rootRouter := mux.NewRouter()
+	rootRouter.SkipClean(true)
```
Comment (jespino) [inline app/server.go]: I'm not 100% sure if this change is backward compatible.

Comment (jupenur) [reply app/server.go]: An ugly fix for backward compatibility would be copying the cleaning code from `mux` and putting it into a middleware that matches all other routes except the email one. But it'd be very ugly.

Comment (mgdelacroix) [reply app/server.go]: After discussing with @jupenur offline, we've decided to keep the fix contained and choose backwards compatibility over solving the unlikely case of emails with quotes and `..`, so I've reverted the commit that applied this change

**Pull request merged**